### PR TITLE
[WIP] DDF-04985 Update yarn dependency list on build

### PR DESCRIPTION
**Pull request by rymach**
branched from https://github.com/codice/ddf/pull/4986
👏 only the last commit is relevant for this pr 👏 

downstream projects often have a requirement to maintain a list of external direct dependencies
updating `yarn-dependency-list.txt` on build assists in this goal

#### Who is reviewing it? 

@gjvera 
@mojogitoverhere 
#### Select relevant component teams: 
@codice/build 
@codice/docs 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.

@adimka
@bdeining

#### How should this be tested?
build and check the deps
#### What are the relevant tickets?
Fixes: #____

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by rymach** [comment]
build now

File: distribution/docs/src/main/resources/content/_reference/_dependencyList/ddf-dependency-list.adoc [hunk 1]
```diff
@@ -3,361 +3,9 @@
...
-* org.xmlunit:xmlunit-matchers:jar:2.5.1
-* us.bpsm:edn-java:jar:0.4.4
-* xalan:serializer:jar:2.7.2
-* xalan:xalan:jar:2.7.2
-* xerces:xercesImpl:jar:2.11.0
-* xerces:xercesImpl:jar:2.9.1
-* xml-apis:xml-apis:jar:1.4.01
-* xpp3:xpp3:jar:1.1.4c
+include::maven-dependency-list.txt[]
+
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/content/_reference/_dependencyList/ddf-dependency-list.adoc]: we may want to have a some sort of header or separator to distinguish the two lists as they are presented alphabetically but it arbitrarily restarts about halfway through.

Maybe drop the period from the end of the `:summary:` property and adding ` - Java` and ` - JavaScript` to the end of the `.{summary} lines.

**Comment by ricklarsen** [comment]
It may not be an issue as I don't think anyone should be distributing artifacts from partial builds, but when I built the docs module to see the results, the script only generated the dependencies from the docs module. Overall, this is a pretty cool improvement to the docs build.

**Comment by rymach** [comment]
closing for now - will revisit later

**Pull request closed without merging**